### PR TITLE
dyn-logger: make sure the logLevel changing log visible always

### PR DESCRIPTION
**Pull request by lxbsz**
Fixes: bz#1661406
Signed-off-by: Xiubo Li <xiubli@redhat.com>

### CODE REVIEW

**Comment by lxbsz** [comment]
The test logs:

```
[2019-03-14 02:28:15.911589] INFO: event->mask: 0x20
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:15.911863] INFO: event->mask: 0x10
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:15.911986] INFO: event->mask: 0x20
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:15.912013] INFO: event->mask: 0x1
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:15.912034] INFO: event->mask: 0x10
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.074942] INFO: event->mask: 0x800
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.076089] INFO: event->mask: 0x4
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.076521] INFO: event->mask: 0x400
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.076919] INFO: event->mask: 0x8000
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.078437] NONE: logLevel now is DEBUG
 [at utils.c+48 :<glusterBlockSetLogLevel>]
[2019-03-14 02:28:23.078994] INFO: event->mask: 0x20
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.079045] INFO: event->mask: 0x1
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:23.079082] INFO: event->mask: 0x10
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:32.350250] INFO: event->mask: 0x20
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:32.350593] INFO: event->mask: 0x10
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:32.350634] INFO: event->mask: 0x20
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:32.350709] INFO: event->mask: 0x1
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:32.350949] INFO: event->mask: 0x10
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:38.693986] INFO: event->mask: 0x800
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:38.696880] INFO: event->mask: 0x4
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:38.696966] INFO: event->mask: 0x400
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:38.697114] INFO: event->mask: 0x8000
 [at dyn-config.c+509 :<glusterBlockDynConfigStart>]
[2019-03-14 02:28:38.697205] NONE: logLevel now is ERROR
 [at utils.c+48 :<glusterBlockSetLogLevel>]


[2019-03-14 02:32:51.334017] NONE: logLevel now is WARNING
 [at utils.c+48 :<glusterBlockSetLogLevel>]
```

**Comment by lxbsz** [comment]
@pkalever @amarts Please review, Thanks.

**Comment by pkalever** [comment]
@lxbsz Can we fix https://bugzilla.redhat.com/show_bug.cgi?id=1661406 in this PR ?

Thanks!

**Comment by pkalever** [comment]
> @lxbsz Can we fix https://bugzilla.redhat.com/show_bug.cgi?id=1661406 in this PR ?
> 
@lxbsz what are your thoughts about this ?

**Comment by lxbsz** [comment]
> > @lxbsz Can we fix https://bugzilla.redhat.com/show_bug.cgi?id=1661406 in this PR ?
> 
> @lxbsz what are your thoughts about this ?

The change here will make sure that the logLevel changing could be seen all the time.

Before the logLevel changing output only be seen when the logLevel is DEBUG, WARN and INFO, because itself is under the INFO logLevel.

And from my last comment you can see that when switching the logLevel from ERROR to any WARN it displayed:

```
[2019-03-14 02:28:38.697205] NONE: logLevel now is ERROR
 [at utils.c+48 :<glusterBlockSetLogLevel>]
[2019-03-14 02:32:51.334017] NONE: logLevel now is WARNING
 [at utils.c+48 :<glusterBlockSetLogLevel>]
```

Isn't that what we need to fix that BZ ?

**Comment by pkalever** [comment]
> Isn't that what we need to fix that BZ ?

My memory is bad, I though we have a plan about syncing gluster-block with fixes in https://github.com/open-iscsi/tcmu-runner/pull/506/commits (namely, 1. directory level watch, 2. checking if we are honoring defaults when commented post changing value, 3. retry configfile open if initial one fails .. etc)

Not sure, If I had discussed this with you or Neha.

**Comment by lxbsz** [comment]
> > Isn't that what we need to fix that BZ ?
> 
> My memory is bad, I though we have a plan about syncing gluster-block with fixes in https://github.com/open-iscsi/tcmu-runner/pull/506/commits (namely, 1. directory level watch, 2. checking if we are honoring defaults when commented post changing value, 3. retry configfile open if initial one fails .. etc)
> 
> Not sure, If I had discussed this with you or Neha.

Sorry Prasanna, I also couldn't remember of this and may miss this discussion. And I am okay with this above.

Thanks.

**Comment by pkalever** [comment]
> Sorry Prasanna, I also couldn't remember of this and may miss this discussion. And I am okay with this above.

Sure @lxbsz, please go ahead and make changes.

Thanks!

**Comment by lxbsz** [comment]
> > Sorry Prasanna, I also couldn't remember of this and may miss this discussion. And I am okay with this above.
> 
> Sure @lxbsz, please go ahead and make changes.
> 
> Thanks!

This is already fixed in https://github.com/gluster/gluster-block/pull/163. Please see https://github.com/gluster/gluster-block/pull/163/commits/94d6d9ab52edc050efe5a41700f53fb26ed229c8

Thanks.

File: utils/utils.c [hunk 1]
```diff
@@ -44,7 +44,7 @@ glusterBlockSetLogLevel(unsigned int logLevel)
   LOCK(gbConf.lock);
   gbConf.logLevel = logLevel;
   UNLOCK(gbConf.lock);
-  LOG("mgmt", GB_LOG_INFO,
+  LOG("mgmt", GB_LOG_NONE,
```
Comment (pkalever) [inline utils/utils.c]: @lxbsz Looks like you are fixing this part in https://github.com/gluster/gluster-block/pull/163/commits/94d6d9ab52edc050efe5a41700f53fb26ed229c8 with GB_LOG_CRIT already. Can we close this PR ?

Comment (lxbsz) [reply utils/utils.c]: Yeah, right.
Since this is duplicated to @163, I will close this PR now.
Thanks Prasanna.

**Pull request closed without merging**